src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Applications/OrganisationStandardVersionOptInHandlerTests/WhenHandlingOrganisationStandardVersionOptInRequest.cs

```diff
@@ -0,0 +1,108 @@
+
+            _sut = new OrganisationStandardVersionOptInHandler(_mockRepository.Object, _mockApplyRepository.Object, _mockStandardRepository.Object,
+            _mockContactQueryRepository.Object, _mockEMailTemplateQueryRepository.Object, _mockMediator.Object,
+            _mockUnitOfWork.Object, _mockLogger.Object);
+        }
+
+        [Test]
+        public async Task ThenOrganisationStandardIsCreated()
+        {
+            var applicationId = Guid.NewGuid();
+
+            var apply = new Domain.Entities.Apply()
+            {
+                Id = applicationId,
+                ApplyData = new ApplyData()
+                {
+                    Apply = new Apply()
+                }
+            };
+
+            var request = new OrganisationStandardVersionOptInRequest()
+            {
+                EndPointAssessorOrganisationId = "ORG",
+                StandardReference = "ST0001",
+                StandardUId = "ST0001_1_2",
+                Version = 1.2M,
+                ApplicationId = applicationId
+            };
+
+            //Arrange
+            _mockRepository.Setup(m => m.GetOrganisationStandardByOrganisationIdAndStandardReference("ORG", "ST0001"))
+                                    .ReturnsAsync(new Domain.Entities.OrganisationStandard() { Id = 123 });
+
+            _mockApplyRepository.Setup(m => m.GetApply(applicationId))
+                                .ReturnsAsync(apply);
+
+            _mockContactQueryRepository.Setup(m => m.GetContactById(It.IsAny<Guid>()))
+                                .ReturnsAsync(new Domain.Entities.Contact() { Email = "a@b.com", DisplayName = "Bob Smith" });
+
+            _mockStandardRepository.Setup(m => m.GetStandardVersionByStandardUId("ST0001_1_2"))
+                                .ReturnsAsync(new Domain.Entities.Standard());
+
+            _mockEMailTemplateQueryRepository.Setup(m => m.GetEmailTemplate(EmailTemplateNames.APPLY_EPAO_RESPONSE))
+                                .ReturnsAsync(new Domain.DTOs.EmailTemplateSummary());
+
+            //Act
+            var result = await _sut.Handle(request, new CancellationToken());
+
+            //Assert
+            _mockRepository.Verify(m => m.CreateOrganisationStandardVersion(It.Is<Domain.Entities.OrganisationStandardVersion>(x =>
+                    x.OrganisationStandardId == 123 &&
+                    x.StandardUId == "ST0001_1_2" &&
+                    x.Version == 1.2M)));
+
+            _mockApplyRepository.Verify(m => m.SubmitApplicationSequence(It.Is<Domain.Entities.Apply>(x =>
+                    x.Id == applicationId &&
+                    x.ApplicationStatus == ApplicationStatus.Approved &&
+                    x.ReviewStatus == ApplicationStatus.Approved &&
+                    x.StandardReference == "ST0001")));
+
+            _mockMediator.Verify(m => m.Send(It.Is<SendEmailRequest>(x => x.Email == "a@b.com"), It.IsAny<CancellationToken>()));
```

Comment:
Can we break the tests apart a bit more please? The verifying email sent should be separate and could also check the contactname and template. Also, could we check that the email is not sent if something proceeding it fails.

Reply:
Fixed